setup.py

```diff
@@ -28,13 +28,29 @@
 
 clang = False
 icc = False
+gcc = True
+
 try:
-    if os.environ['CC'] == "clang":
-        clang = True
+    if os.environ['CC'] == "gcc":
+        gcc = True
+    elif os.environ['CC'] != "":
+        gcc = False
 except KeyError:
     pass
 
-if not clang:
+
+if not gcc:
+    try:
+        if os.environ['CC'] == "clang":
+            # note, the conda provideed clang is not detected here
+            # and this is on purpose, as MacOS clang is very different
+            # than conda-provised one (which is llvm based)
+            clang = True
+        elif os.environ['CC'] == "icc":
+            icc = True
```

Comment:
```suggestion
        elif "icc" in os.environ['CC']:
            icc = True
```

Reply:
Using just
`"icc" in` 
is too simplistic
It could match a path name that contains icc

Reply:
Added better compiler detection.
Should do what the above was supposed to achieve.